### PR TITLE
unique execution role

### PR DESCRIPTION
**Pull request by abdullahaslam306**

### CODE REVIEW

File: plugins/aws/lambda/lambdaUniqueExecutionRole.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            for (var lambdaFunc of listFunctions.data) {
+                if (!lambdaFunc.FunctionArn) continue;
+                
+                if (executionRoleList.includes(lambdaFunc.Role)){
+                    helpers.addResult(results, 2, 'Lambda function does not have unique execution role', region, lambdaFunc.FunctionArn);
+                } else {
+                    executionRoleList.push(lambdaFunc.Role);
+                    helpers.addResult(results, 0, 'Lambda function have unique execution role', region, lambdaFunc.FunctionArn);
+                }
+            }
```
Comment (alphadev4) [inline plugins/aws/lambda/lambdaUniqueExecutionRole.js:47]: isnt that conflicting, if two lambda functions are sharing same execution role, both of lambda's should be producing false result?

File: plugins/aws/lambda/lambdaUniqueExecutionRole.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+            let executionRoleList = [];
+            for (var lambdaFunc of listFunctions.data) {
+                if (!lambdaFunc.FunctionArn) continue;
+                
+                if (executionRoleList.includes(lambdaFunc.Role)){
+                    helpers.addResult(results, 2, 'Lambda function does not have unique execution role', region, lambdaFunc.FunctionArn);
+                } else {
+                    executionRoleList.push(lambdaFunc.Role);
+                    helpers.addResult(results, 0, 'Lambda function have unique execution role', region, lambdaFunc.FunctionArn);
```
Comment (alphadev4) [inline plugins/aws/lambda/lambdaUniqueExecutionRole.js:45]: isnt that conflicting, if two lambda functions are sharing same execution role, both of lambda's should be producing false result?

File: plugins/aws/lambda/lambdaUniqueExecutionRole.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                }
```
Comment (alphadev4) [inline plugins/aws/lambda/lambdaUniqueExecutionRole.js:46]: isnt that conflicting, if two lambda functions are sharing same execution role, both of lambda's should be producing false result?

File: plugins/aws/lambda/lambdaUniqueExecutionRole.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+                helpers.addResult(results, 0, 'No Lambda functions found', region);
+                return rcb();
+            }
+            let roleMap = {};
+            for (let lambdaFunc of listFunctions.data) {
+                if (roleMap[lambdaFunc.Role]){
+                    roleMap[lambdaFunc.Role].push(lambdaFunc.FunctionArn);
+                } else {
+                    roleMap[lambdaFunc.Role] = [lambdaFunc.FunctionArn];
+                }
```
Comment (AkhtarAmir) [inline plugins/aws/lambda/lambdaUniqueExecutionRole.js:42]:
```suggestion
                if (!lambdaFunc.Role) continue;

                if (roleMap[lambdaFunc.Role]){
                    roleMap[lambdaFunc.Role].push(lambdaFunc.FunctionArn);
                } else {
                    roleMap[lambdaFunc.Role] = [lambdaFunc.FunctionArn];
                }
```

File: plugins/aws/lambda/lambdaUniqueExecutionRole.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'Lambda Unique Execution Role',
+    category: 'Lambda',
+    domain: 'Serverless',
+    description: 'Ensure that AWS Lambda functions do not share the same execution role',
```
Comment (AkhtarAmir) [inline plugins/aws/lambda/lambdaUniqueExecutionRole.js:8]:
```suggestion
    description: 'Ensure that AWS Lambda functions do not share the same execution role.',
```

**Pull request merged**